Watchman.Discord/Areas/Users/Controllers/WarnsController.cs

```diff
@@ -40,10 +40,18 @@ public async Task AddWarn(AddWarnCommand command, Contexts contexts)
         {
             var messageService = this._messagesServiceFactory.Create(contexts);
             var mentionedUser = await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
-            await this._warnService.AddWarnToUser(command, contexts, mentionedUser);
+            await this._warnService.AddWarnToUser(contexts.User.Id, mentionedUser.Id, command.Reason, contexts.Server.Id);
             await messageService.SendResponse(x => x.UserHasBeenWarned(contexts.User.Name, mentionedUser.Name, command.Reason));
         }
 
+        [AdminCommand]
+        public async Task ResetWarns(ResetWarnsCommand command, Contexts contexts)
+        {
+            await this._warnService.RemoveUserWarns(command.User, contexts.Server.Id);
+            var messageService = this._messagesServiceFactory.Create(contexts);
+            await messageService.SendResponse(x => x.AllWarnsRemovedFromUser(contexts.User.Id, command.User));
```

Comment:
zrobiłbym tak że komenda sama w sobie zwraca informacje `czy na pewno chcesz to zrobić`
i potrzebne byłoby użycie parametru `-force` 

no i też fajnie gdyby dało sie łatwo przywrócić usunięte warny 
żeby nie było akcji, że komuś przypadkiem sie "źle kliknie"

Reply:
> no i też fajnie gdyby dało sie łatwo przywrócić usunięte warny
To chyba zalatwia teraz fakt że mamy tylko IsDeleted = true, czy coś do tego jeszcze trzeba dorobić?

Reply:
w sensie
chodzi mi o to żeby nie było sytuacji typu ktoś przy wybieraniu użytkownika źle kliknie

w przypadku np mutowania można łatwo kogoś odmutować
ale co w przypadku usuwania warnów?

tak w sumie to teraz myśle czy usuwanie warnów ma w ogóle sens 
teoretycznie powinno sie dać
ale w praktyce - może być błąd o ktoś może sam sobie usunąć warny 

a dodatkowo dobrze byłoby mieć wgląd w czyjąś historie mimo wszystko
a nie że ktoś był wkurzający, zostaje kolegą admina i nagle jest "czysty"

Reply:
w sumie to resetowanie można wywalić albo zakomentować - i ewentualnie sie odkomentuje
@gagyn co myślisz?

Reply:
ja bym zostawił dodanie resetowania na przyszłość, w sensie warto tu by było jakoś tak zrobić, że warna się usuwa po ID albo tylko po przez API
ale jakaś forma usuwania powinna być, bo z drugiej strony ktoś mógł źle kliknąć i ktoś dostał warna, a nie powinien
może trzeba dwa razy wpisać komendę? wtedy mniejsza szansa na pomyłkę

Reply:
@Marcin99b @gagyn To jak robimy?

Reply:
fajnie by było zrobić coś na zasadzie:
-removelastwarn @userZwarnem
tutaj przydałoby się gdyby były już reakcje, bo bot mógłby wysłać jakąś odpowiedź i dopiero po potwierdzeniu reakcją usuwałby warna
ale teraz oczywiście nie musisz tego robić
@Marcin99b co myślisz?

Reply:
zatwierdzanie reakcją byłoby ok
aktualnie można zrobić usuwanie pojedynczych na podstawie ID warna
ale teraz pytanie -> skąd to ID brać
wyświetlanie guidów obok listy nie będzie wyglądać dobrze
może generowanie tymczasowych id to lepsza opcja

Reply:
Wydaje mi się że to już by było dość duże na innego taska, więc wolałbym tu się ograniczyć do "Czy rezygnujemy z tego co tu jest?" i jeśli nie rezygnujemy, to wrzucić to resetowanie warnów dalej, a jak rezygnujemy to odciąć dostęp dla usera i najwyżej kiedyś ktoś dorobi wszystko co jeszcze musi pośredniczyć aby można było resetować warny.

Reply:
o mam pomysł jak to łatwo zrobić
kiedy wyśle się tą komendę do usunięcia warna (można usuwać tylko jednego warna na raz przy jednym wywołaniu)
to bot zwróci komunikat czy na pewno chcesz usunąć warna i niech wyśle pierwszy znak z Guid tego Warna
i potem trzeba wysłać drugi raz komendę + z dopisanym tym znakiem

Reply:
Ok, to zrobię takie coś